docs/testing.md

```diff
@@ -0,0 +1,136 @@
+
+On the other hand, many changes will involve a tighter integration with some dependency that makes us test it at the edges of the dependency interface. We have some support for that as well.
+
+### Kubernetes
+
+For Kubernetes, the [operator-runtime library has a fake package](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client/fake) that allows you to create a client.Client using [`NewClientBuilder`](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client/fake#NewClientBuilder) and fake many Kubernetes interactions. Just note the limitations:
+
+- You must pre-set the relevant Kubernetes status in the client before each test.
+- This fake client _cannot simulate complex Kubernetes status scenarios_. But remember your unit tests should be focused on a single check, not validating a full workflow or step sequence.
+- This fake client does not support error injection. You might want to resort to TDD and use simple functions for that instead.
+
+### Atlas
+
+For Atlas, the operator currently uses the [go-client-mongodb-atlas](https://github.com/mongodb/go-client-mongodb-atlas#go-client-mongodb-atlas). This [client main struct is composed of a set of interfaces](https://pkg.go.dev/go.mongodb.org/atlas@v0.31.0/mongodbatlas#Client), one per Service behind Atlas. A simple way to mock such a client for unit tests, that will probably only call a couple of API endpoints from Atlas at a time, is to replace those service implementations by the mock:
+
+- The mock is an struct that implements the Service interface by calling methods set in that struct for each of the interface methods defined. [See sample below](#sample-snippets).
+- Under `test-support/atlastest` you should find all mocked services already being unit tested using this approach.
+- And from them, you can also [find examples of how they are being used in unit tests](#sample-usage-mock).
+- If a service is mocked but the method you need is not configurable in the mock struct, it means it has not been used before in a test.
+- You can add service method implementations following the same structure as the ones already in place. [Add a new function to be set in the struct and called in the method, and just remember to remove the context first parameter](#sample-projects-mock), ignored in this mocks.
+- If a service is not mocked yet, create a new entry for it at `test-support/atlastest` following the same conventions as already present service mocks there.
+- You can use the [impl tool](https://github.com/josharian/impl) to generate the skeleton implementation. Alternativaly, you can try using your IDE such for filling the implemntation with “unimplemented” method calls so that the service interface is satisfied for the compiler. After that, just fill in the methods needed for the test and ignore the rest.
```

Comment:
```suggestion
- You can use the [impl tool](https://github.com/josharian/impl) to generate the skeleton implementation. Alternativaly, you can try using your IDE for filling the implementation with “unimplemented” method calls so that the service interface is satisfied for the compiler. After that, just fill in the methods needed for the test and ignore the rest.
```